src/node_env_var.cc

```diff
@@ -66,89 +66,76 @@ std::shared_ptr<KVStore> system_environment = std::make_shared<RealEnvStore>();
 Local<String> RealEnvStore::Get(Isolate* isolate,
                                 Local<String> property) const {
   Mutex::ScopedLock lock(per_process::env_var_mutex);
-#ifdef __POSIX__
+
   node::Utf8Value key(isolate, property);
-  const char* val = getenv(*key);
-  if (val) {
-    return String::NewFromUtf8(isolate, val, NewStringType::kNormal)
-        .ToLocalChecked();
-  }
-#else  // _WIN32
-  node::TwoByteValue key(isolate, property);
-  WCHAR buffer[32767];  // The maximum size allowed for environment variables.
-  SetLastError(ERROR_SUCCESS);
-  DWORD result = GetEnvironmentVariableW(
-      reinterpret_cast<WCHAR*>(*key), buffer, arraysize(buffer));
-  // If result >= sizeof buffer the buffer was too small. That should never
-  // happen. If result == 0 and result != ERROR_SUCCESS the variable was not
-  // found.
-  if ((result > 0 || GetLastError() == ERROR_SUCCESS) &&
-      result < arraysize(buffer)) {
-    const uint16_t* two_byte_buffer = reinterpret_cast<const uint16_t*>(buffer);
-    v8::MaybeLocal<String> rc = String::NewFromTwoByte(
-        isolate, two_byte_buffer, NewStringType::kNormal);
-    if (rc.IsEmpty()) {
-      isolate->ThrowException(ERR_STRING_TOO_LONG(isolate));
+  char* val = nullptr;
+  size_t init_sz = 256;
+
+  // Allocate 256 bytes initially, if not enough reallocate.
+  val = Malloc(init_sz);
+
+  int ret = uv_os_getenv(*key, val, &init_sz);
+
+  if (UV_ENOBUFS == ret) {
+    // Buffer is not large enough, reallocate to the updated init_sz
+    // and fetch env value again.
+    val = Realloc(val, init_sz);
+
+    ret = uv_os_getenv(*key, val, &init_sz);
+
+    // Still failed to fetch env value return emptry string.
+    if (UV_ENOBUFS == ret || UV_ENOENT == ret) {
       return Local<String>();
     }
-    return rc.ToLocalChecked();
+  } else if (UV_ENOENT == ret) {
+    return Local<String>();
   }
-#endif
-  return Local<String>();
+
+  MaybeLocal<String> value_string =
+      String::NewFromUtf8(isolate, val, NewStringType::kNormal);
+
+  if (value_string.IsEmpty()) {
+       isolate->ThrowException(ERR_STRING_TOO_LONG(isolate));
```

Comment:
The indentation here is off (we indent by 2 spaces here)

Reply:
fixed @ https://github.com/nodejs/node/pull/27310/commits/4bc0fb4880cf7254118b64752547b3b5c8023c24